### ISSUE TITLE
Error for empty vpc_security_group_ids block is unhelpful

### ISSUE THREAD

**Issue by kevinm416** [hashicorp/terraform#2293]
For an instance like this:

```
resource "aws_instance" "instance-1" {
    ami = "ami-e7527ed7"
    instance_type = "t2.micro"
    key_name = "key"
    subnet_id = "${aws_subnet.subnet-1.id}"
    vpc_security_group_ids = [
    ]
}
```

You get the error message:
```
$ ../terraform-install/terraform apply 
aws_ebs_volume.block-store-1: Refreshing state... (ID: vol-90e67080)
aws_vpc.migrage-storage-environment: Refreshing state... (ID: vpc-75ee6410)
aws_subnet.subnet-1: Refreshing state... (ID: subnet-511d7034)
aws_internet_gateway.internet-gateway-1: Refreshing state... (ID: igw-76a30b13)
aws_route_table.main-route-table: Refreshing state... (ID: rtb-bb1a84de)
aws_instance.instance-1: Refreshing state... (ID: i-5e82fba9)
aws_main_route_table_association.main-route-table-association: Refreshing state... (ID: rtbassoc-145c3771)
aws_eip.instance-1-ip-address: Refreshing state... (ID: eipalloc-dd3e96b8)
aws_instance.instance-1: Modifying...
  vpc_security_group_ids.#: "1" => "0"
aws_instance.instance-1: Error: 1 error(s) occurred:

* InvalidParameterCombination: No attributes specified.
	status code: 400, request id: []
Error applying plan:

1 error(s) occurred:

* 1 error(s) occurred:

* 1 error(s) occurred:

* InvalidParameterCombination: No attributes specified.
	status code: 400, request id: []

Terraform does not automatically rollback in the face of errors.
Instead, your Terraform state file has been partially updated with
any resources that successfully completed. Please address the error
above and apply again to incrementally change your infrastructure.
```

I think the error message should be something along the lines of: "vpc_security_group_ids cannot be empty for instance <instance_name>". Or, terraform could deal with an empty vpc_security_group_ids block.

**Comment by phinze** [comment]
Thanks for the report - fixed in #2311 so we will just treat the empty list as though there was nothing specified.

**Comment by anxiousmodernman** [comment]
This is a heads up. Error message could still be better, or perhaps there's an error in the `apply` phase vs the parse/`plan` phase. 

Got a similar error when I specified `vpc_security_group_ids = []` under my instance declaration. After running `terraform apply`:

```
* aws_instance.pub_networktest: InvalidParameterCombination: No attributes specified.
        status code: 400, request id: 1ddee362-3379-403a-8c51-6107593078d3
```

This error was repeated for another instance, but here is an exmaple:

```hcl
resource "aws_instance" "pub_networktest" {
  ami                         = "ami-1d407f0a"
  availability_zone           = "${element(var.availability_zones, 0)}"
  instance_type               = "t2.micro"
  key_name                    = "${var.key_name}"                               
  subnet_id                   = "${var.public_subnet_id}"
  associate_public_ip_address = true
  vpc_security_group_ids = []

  tags {
    Name = "pub_networktest"
  }
}
```
`terraform plan` gave the thumbs up, but `terraform apply` threw the error at runtime.

In the end, I did want to specify a security group ID in this block, and adding an ID made the `apply` work.

Terraform version `v0.7.11`
